pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/SegmentCreationJob.java

```diff
@@ -190,7 +199,31 @@ public void run() throws Exception {
   }
 
   protected void setAdditionalJobProperties(Job job) throws Exception {
+    List<PushLocation> pushLocations = new ArrayList<>();
+
+    for (String host : _hosts) {
+      pushLocations.add(new PushLocation.PushLocationBuilder()
+          .setHost(host)
+          .setPort(_port)
+          .build());
+    }
+
+    ControllerRestApi controllerRestApiObject = new ControllerRestApi(pushLocations, _tableName);
+
+    TableConfig tableConfig = controllerRestApiObject.getTableConfig();
```

Comment:
why do we have to do this, why cant we set the entire table config. This is adding unnecessary constants for every field in table config

Reply:
In the next one, I plan to add a new constructor in SegmentGeneratorConfig that takes in table config and sets parameters (startree, etc, so we don't have to serializet he whole thing). However, I like the extra logic/control that I feel should not go into the constructor for this parameter in particular in particular because I want the segment names to be correct.